src/pt/huntersscans/src/eu/kanade/tachiyomi/extension/pt/huntersscans/HuntersScans.kt

```diff
@@ -1,23 +1,247 @@
+
+        val origin = response.request.url
+        var nextPage = 2
+
+        do {
+            if (chapters.size < 2) { break }
+
+            chapters += fetchChapterRequest(origin, nextPage).sortedBy { it.name.toInt() }
+
+            alwaysVisibleChapters += chapters.removeFirst()
+            alwaysVisibleChapters += chapters.removeLast()
+
+            nextPage++
+        }
+        while (!containsDuplicate(chapters) && chapters.isNotEmpty())
+
+        chapters += alwaysVisibleChapters
+
+        return chapters
+            .distinctBy { it.name }
+            .sortedBy { it.name.toInt() }
+            .reversed()
+    }
+
+    override fun chapterListSelector() = "script"
+
+    override fun imageUrlParse(document: Document) = ""
+
+    override fun latestUpdatesFromElement(element: Element) =
+        SManga.create().apply {
+            title = element.selectFirst("h3")!!.ownText()
+            thumbnail_url = element.selectFirst("img")?.absUrl("src")
+            setUrlWithoutDomain(element.selectFirst("a")!!.absUrl("href"))
+        }
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun fetchLatestUpdates(page: Int): Observable<MangasPage> =
+        super.fetchLatestUpdates(page).map { it.removeNovels() }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$baseUrl/ultimas-atualizacoes".toHttpUrl().newBuilder()
+            .addQueryParameter("page", "$page")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesSelector() = "main > div div:nth-child(2) > div.relative"
+
+    override fun mangaDetailsParse(document: Document) =
+        SManga.create().apply {
+            val container = document.selectFirst("div.container")!!
+            title = container.selectFirst("h2")!!.ownText()
+            thumbnail_url = container.selectFirst("img")?.absUrl("src")
+            genre = container.select("ul > li:nth-child(5) p").joinToString { it.ownText() }
+
+            val statusLabel = container.selectFirst("ul > li:nth-child(3) p")?.ownText()
+            status = statusManga.getOrDefault(statusLabel, SManga.UNKNOWN)
+            description = document.selectFirst("main > div > div")?.text()
+        }
+
+    override fun pageListParse(document: Document) =
+        document.select("main.container img")
+            .mapIndexed { i, element -> Page(i, imageUrl = element.absUrl("src")) }
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        title = element.selectFirst("h2")!!.ownText()
+        thumbnail_url = element.selectFirst("img")?.absUrl("src")
+        setUrlWithoutDomain(element.absUrl("href"))
+    }
+
+    override fun popularMangaNextPageSelector() = "li[aria-label='next page button']:not([aria-disabled])"
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> =
+        super.fetchPopularManga(page).map { it.removeNovels() }
```

Comment:
move this to `popularMangaParse` as well

Reply:
Resolved: https://github.com/keiyoushi/extensions-source/pull/1650/commits/d074c6330e3bf59ae82d36dbe0bfaa0d42f3392b